### PR TITLE
Fix PMD warnings from Java “Best Practices” rule set

### PR DESCRIPTION
**Pull request by alexbishop1**
Fix PMD warnings from Java “Best Practices” rule set (except for those rules we exclude — this PR adds a couple of new exclusions).

See individual commits for details.

### CODE REVIEW

File: src/main/java/uk/gov/pay/adminusers/app/config/AdminUsersModule.java [hunk 1]
```diff
@@ -96,7 +96,7 @@ private JpaPersistModule jpaModule(AdminUsersConfig configuration) {
     }
 
     @Provides
-    private NotificationService provideUserNotificationService() {
+    public NotificationService provideUserNotificationService() {
```
Comment (oswaldquek) [inline src/main/java/uk/gov/pay/adminusers/app/config/AdminUsersModule.java]: It seems the private keyword on the method didn't stop the app from running...

File: src/main/java/uk/gov/pay/adminusers/model/PatchRequest.java [hunk 2]
```diff
@@ -26,6 +26,10 @@ public static PatchRequest from(JsonNode node) {
         return new PatchRequest(node.get("op").asText(), node.get("path").asText(), node.get("value").asText());
     }
 
+    public String getOp() {
+        return op;
+    }
+
```
Comment (oswaldquek) [inline src/main/java/uk/gov/pay/adminusers/model/PatchRequest.java]: what breakage does adding this getter method fix?

File: src/main/java/uk/gov/pay/adminusers/app/config/AdminUsersModule.java (hunk 1 above)
Comment (alexbishop1) [reply src/main/java/uk/gov/pay/adminusers/app/config/AdminUsersModule.java]: It stops PMD complaining about an apparently unused private method.

File: src/main/java/uk/gov/pay/adminusers/model/PatchRequest.java (hunk 2 above)
Comment (alexbishop1) [reply src/main/java/uk/gov/pay/adminusers/model/PatchRequest.java]: It stops PMD warning about an unused private field.

**Pull request merged**